flink-ml-lib/src/test/java/org/apache/flink/ml/feature/stringindexer/IndexToStringModelTest.java

```diff
@@ -0,0 +1,162 @@
+                Arrays.asList("inputCol1", "inputCol2", "outputCol1", "outputCol2"),
+                output.getResolvedSchema().getColumnNames());
+    }
+
+    @Test
+    public void testIllegalInput() {
+        IndexToStringModel indexToStringModel =
+                new IndexToStringModel()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setModelData(modelTable);
+        Table output = indexToStringModel.transform(illegalPredictTable)[0];
+        try {
+            IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+            fail();
+        } catch (Exception e) {
+            assertEquals(
+                    "The input contains unseen index: 4.",
+                    e.getCause().getCause().getCause().getCause().getCause().getMessage());
+        }
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testPredict() throws Exception {
+        IndexToStringModel indexToStringModel =
+                new IndexToStringModel()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setModelData(modelTable);
+        Table output = indexToStringModel.transform(predictTable)[0];
+        List<Row> predictedResult =
+                IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        StringIndexerTest.verifyPredictionResult(expectedPrediction, predictedResult);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testSaveLoadAndPredict() throws Exception {
+        IndexToStringModel model =
+                new IndexToStringModel()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setModelData(modelTable);
+        model = StageTestUtils.saveAndReload(env, model, tempFolder.newFolder().getAbsolutePath());
+        assertEquals(
+                Collections.singletonList("stringArrays"),
+                model.getModelData()[0].getResolvedSchema().getColumnNames());
+        Table output = model.transform(predictTable)[0];
+        List<Row> predictedResult =
+                IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        StringIndexerTest.verifyPredictionResult(expectedPrediction, predictedResult);
+    }
+
+    @Test
+    public void testGetModelData() throws Exception {
+        IndexToStringModel model =
+                new IndexToStringModel()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setModelData(modelTable);
+
+        StringIndexerModelData modelData =
+                StringIndexerModelData.getModelDataStream(model.getModelData()[0])
+                        .executeAndCollect()
```

Comment:
Should we replace `next()` with `IteratorUtils.toList(...)` for consistency with other existing tests?

This approach would also verify that there is exactly one record in the model data stream in this test.

Same for `StringIndexerTest`.